### PR TITLE
Upgrade provider to Terraform 0.12 SDK

### PR DESCRIPTION
**Pull request by alexsomesan**
This change upgrades the Terraform SDK to make the provider compatible with the upcoming Terraform 0.12.
It also makes the necessary adaptations to accommodate HCL2 in test configurations and the breaking changes introduced in Terraform 0.12.

### CODE REVIEW

File: kubernetes/resource_kubernetes_namespace_test.go [hunk 1]
```diff
@@ -111,21 +111,23 @@ func TestAccKubernetesNamespace_basic(t *testing.T) {
...
-		Steps: []resource.TestStep{
-			{
-				Config:      testAccKubernetesNamespaceConfig_invalidLabelValueType(nsName),
-				ExpectError: regexp.MustCompile("Expected value to be string"),
-			},
-		},
-	})
-}
+// Test disabled: this seems to no longer be an error in Terraform 0.12
+// TODO: Confirm with core team.
```
Comment (katbyte) [inline kubernetes/resource_kubernetes_namespace_test.go]: Still an applicable todo?

File: kubernetes/resource_kubernetes_namespace_test.go [hunk 2]
```diff
@@ -430,18 +431,19 @@ resource "kubernetes_namespace" "test" {
...
-      "integer" = 2
-      "bool"    = true
-    }
-
-    name = "%s"
-  }
-}
-`, nsName)
-}
+// Test disabled: this seems to no longer be an error in Terraform 0.12
```
Comment (katbyte) [inline kubernetes/resource_kubernetes_namespace_test.go]: as above

File: kubernetes/resource_kubernetes_namespace_test.go (hunk 1 above)
Comment (alexsomesan) [reply kubernetes/resource_kubernetes_namespace_test.go]: Good point. I missed this one.
Let me clarify that.

Comment (alexsomesan) [reply kubernetes/resource_kubernetes_namespace_test.go]: Done. Clarified with Core that this not relevant anymore (everything turns into a string automatically).

File: kubernetes/resource_kubernetes_namespace_test.go (hunk 2 above)
Comment (alexsomesan) [reply kubernetes/resource_kubernetes_namespace_test.go]: Done, removed.

**Comment by alexsomesan** [comment]
CI is green. Merging... 🙏

**Pull request merged**